test/test_torch.py

```diff
@@ -955,6 +955,31 @@ def test_where_bool_tensor(self):
             res = torch.where(a > 0)
             self.assertEqual(1, len(res))
 
+    def test_where_tensor_cpu(self):
```

Comment:
if we're mainly testing for the promotion cases, this might make more sense to include in test_type_promotion.py. We may also want to add `torch.where` to the lists of operators tested by some of the other existing promotion tests